packages/react/src/core/ContextMenu.tsx

```diff
@@ -17,44 +19,80 @@ interface Props extends ComponentProps<'div'> {
   /** Callback invoked when file is changed. */
   onFileChange?: (event: FileChangeEvent | FileRenameEvent) => void;
 
+  /** Glob patterns for paths that allow editing files and folders. Defaults to `['**']`. */
+  allowEditPatterns?: string[];
+
   /** Directory of the clicked file. */
   directory: string;
 
   /** Whether to render new files/directories before or after the trigger element. Defaults to `'before'`. */
   position?: 'before' | 'after';
 
   /** Localized texts for menu. */
-  i18n?: Pick<I18n, 'fileTreeCreateFileText' | 'fileTreeCreateFolderText'>;
+  i18n?: Pick<
+    I18n,
+    | 'fileTreeCreateFileText'
+    | 'fileTreeCreateFolderText'
+    | 'fileTreeFailedToCreateFileText'
+    | 'fileTreeFailedToCreateFolderText'
+    | 'fileTreeAllowedPatternsText'
+  >;
 
   /** Props for trigger wrapper. */
   triggerProps?: ComponentProps<'div'> & { 'data-testid'?: string };
 }
 
+const i18nDefaults = {
+  fileTreeFailedToCreateFileText: 'Failed to create file "${filename}".',
+  fileTreeFailedToCreateFolderText: 'Failed to create folder "${filename}".',
+  fileTreeCreateFileText: 'Create file',
+  fileTreeCreateFolderText: 'Create folder',
+  fileTreeAllowedPatternsText: 'Allowed patterns are:',
+} as const satisfies Props['i18n'];
```

Comment:
As React components can be used without TutorialKit, we need to store copies of localizations in these components as well. We might want to move those default localizations in separate package, or serve defaults from `@tutorialkit/types` maybe. 🤔

Reply:
Oh that's a good point, I think having them in `@tutorialkit/types` make sense.